emily_cron/app/clients/mempool.py

```diff
@@ -7,11 +7,48 @@
 logger = logging.getLogger(__name__)
 
 
+def _collect_rbf_txids(data: dict[str, Any] | None) -> set[str]:
+    """Recursively collect all RBF transaction IDs from the replacement chain.
+
+    Args:
+        data: Transaction replacement data from mempool API
+
+    Returns:
+        Set of transaction IDs that replaced the original transaction
+    """
+    txids = set()
+
+    if data is None:
+        return txids
+
+    if tx := data.get("tx", {}):
+        if txid := tx.get("txid"):
+            txids.add(txid)
+
+    for replacement in data.get("replaces", []):
+        txids.update(_collect_rbf_txids(replacement))
+
+    return txids
+
+
 class MempoolAPI(APIClient):
     """Client for interacting with the Mempool API."""
 
     BASE_URL = settings.MEMPOOL_API_URL
 
+    @classmethod
+    def check_for_rbf(cls, txid: str) -> set[str]:
```

Comment:
Is `cls` an alias for `self` ??

Reply:
Kinda, it's shorthand for class. This is a class method, so it could be called without an instance of the type. When calling this method, it does not have access to the properties that get set during `__init__` or other methods, it can only access class fields.